src/prelude.rs

```diff
@@ -447,6 +449,9 @@ impl Default for OperationStepDecision {
     }
 }
 
+// NOTE.2022-11-04 not very pretty singleton like global variable
+pub static CONFIG_DIR: Mutex<Option<PathBuf>> = Mutex::new(None);
+
 pub fn app_dir() -> std::path::PathBuf {
     if let Ok(mut flag) = std::env::current_exe() {
```

Comment:
When `CONFIG_DIR` is `Some`, it should take priority over the portable flag file. We can also delay the clone:

```rust
pub fn app_dir() -> std::path::PathBuf {
    if let Some(dir) = CONFIG_DIR.lock().unwrap().as_ref() {
        return dir.clone();
    }

    if let Ok(mut flag) = std::env::current_exe() {
        flag.pop();
        flag.push(PORTABLE_FLAG_FILE_NAME);
        if flag.exists() {
            flag.pop();
            return flag;
        }
    }

    let mut path = dirs::config_dir().unwrap();
    path.push(APP_DIR_NAME);
    path
}
```